integration_tests/test_basic.py

```diff
@@ -101,3 +100,55 @@ def test_native_call(cronos):
     )
     receipt = w3.eth.wait_for_transaction_receipt(txhash)
     assert receipt.status == 0, "should fail"
+
+def test_statesync(cronos):
+    ## cronos fixture
+    # Load cronos-devnet.yaml
+    # Spawn pystarport with the yaml, port 26650 (multiple nodes will be created based on `validators`)
+    # Return a Cronos object (Defined in network.py)
+
+    ## do some transactions
+    from_addr = "crc1q04jewhxw4xxu3vlg3rc85240h9q7ns6hglz0g"
+    to_addr = "crc16z0herz998946wr659lr84c8c556da55dc34hh"
+    coins = "10basetcro"
+    node = cronos.node_rpc(0)
+    txhash_1 = cronos.cosmos_cli(0).transfer(from_addr, to_addr, coins)["txhash"]
+    # discovery_time is set to 5 seconds, add extra seconds for processing
+    wait_for_block(cronos.cosmos_cli(0), 15)
```

Comment:
Do we wait for the block until height 15?
https://github.com/crypto-org-chain/cronos/blob/94002c55f289af3e1b4ce4d6182882594aea40aa/integration_tests/utils.py#L44

Reply:
maybe we can use query_tx bound with a timeout to confirm the transaction has been processed into the chain.
it can shorten the time of the test.

Reply:
`wati_for_block` is not perfect and it not time consuming in a small test.

By the way, I think in all cosmos cli commands would not have timeout bound, it is a general issue, not just in `query_tx`. 

Any other good solutions?

Reply:
It is ok to not wait to height 15 here, but it is still necessary to wait to height 20 (or less, it really depends) at https://github.com/damoncro/cronos/blob/test_statesync_with_cronos_fixture/integration_tests/test_basic.py#L133. 

The problem is how many timeout we should bound?  I guess the timeout varies in different machines.

Here the test is to make sure the "catching_up" is "False" after height 20.

Reply:
Could we use web3's `wait_for_transaction_receipt`?

Reply:
Oh, I just found that cronos tx hash can not query by eth api.

Reply:
Okay, maybe you can add a comment to explain why the test waits for 20 blocks. Or perhaps you can check `catching_up` status with a time out?

Reply:
@JayT106 `wait_for_block(cronos.cosmos_cli(0), 15)` is also to wait it generate some empty blocks, so that we can start from height 16 in statesync node.